api/src/api/application_alpha/application_route.py

```diff
@@ -491,3 +499,47 @@ def application_audit_list(
     return response.ApiResponse(
         message="Success", data=audit_events, pagination_info=pagination_info
     )
+
+
+@application_blueprint.post("/applications/<uuid:application_id>/submissions")
+@application_blueprint.input(ApplicationSubmissionsRequestSchema())
+@application_blueprint.output(ApplicationSubmissionsResponseSchema())
+@application_blueprint.doc(
+    responses=[200, 401, 403, 404], security=jwt_or_user_api_key_security_schemes
+)
+@jwt_or_user_api_key_multi_auth.login_required
+@flask_db.with_db_session()
+def application_submissions_list(
+    db_session: db.Session, application_id: UUID, json_data: dict
+) -> response.ApiResponse:
+    """Get all application submissions for an application"""
+    add_extra_data_to_current_request_logs({"application_id": application_id})
+    logger.info("POST /alpha/applications/:application_id/submissions")
+
+    # Get user from the multi-auth (supports both JWT and User API Key)
+    multi_auth_user = jwt_or_user_api_key_multi_auth.get_user()
+
+    if isinstance(multi_auth_user.user, UserTokenSession):
+        user = multi_auth_user.user.user
+    elif isinstance(multi_auth_user.user, UserApiKey):
+        user = multi_auth_user.user.user
+    else:
+        raise Exception(f"Unknown user type: {type(multi_auth_user.user)}")
```

Comment:
Could you add a note to adjust this after we adjust the auth (see my other comment)?